tests/component/test_task_resource_warnings.py

```diff
@@ -0,0 +1,45 @@
+import warnings
+import nidaqmx
+
+import pytest
+
+from nidaqmx.errors import DaqResourceWarning
+
+
+@pytest.mark.library_only
+def test__unclosed_task__leak_task__resource_warning_raised(task):
+    """Test to validate unclosed tasks."""
+    interpreter = task._interpreter
+    task_handle = task._handle
```

Comment:
Sorry, I suggested this before I realized that we can still close the task after calling `task.__del__()`.

The `task` fixture owns the task and automatically closes it, so delete the interpreter/task_handle stuff. Calling `interpreter.clear_task()` followed by `task.clear()` is a double clear.

For the 1st two test functions that use the `task` fixture, all you need is the "act" section (`catch_warnings` and `__del__`). The 3rd test function that uses `init_kwargs` is fine.